libs/MeshKernel/src/AveragingInterpolation.cpp

```diff
@@ -65,103 +65,84 @@ void AveragingInterpolation::Compute()
     // build sample r-tree for searches
     m_samplesRtree.BuildTree(m_samples);
 
-    auto interpolatedResults = ComputeOnLocations();
+    std::vector<Point> dualFacePolygon;
+    m_nodeResults.resize(m_mesh->GetNumNodes(), constants::missing::doubleValue);
+    std::fill(m_nodeResults.begin(), m_nodeResults.end(), constants::missing::doubleValue);
+
+    // make sure edge centers are computed
+    m_mesh->ComputeEdgesCenters();
+    for (size_t n = 0; n < m_mesh->GetNumNodes(); ++n)
+    {
+        m_mesh->MakeDualFace(n, m_relativeSearchRadius, dualFacePolygon);
+
+        const auto result = ComputeOnPolygon(dualFacePolygon, m_mesh->m_nodes[n]);
+
+        // flag the visited samples
+        for (size_t i = 0; i < m_samplesRtree.GetQueryResultSize(); ++i)
+        {
+            const auto sample = m_samplesRtree.GetQueryResult(i);
+            m_visitedSamples[sample] = true;
+        }
+
+        m_nodeResults[n] = result;
+    }
 
     // for edges, an average of the nodal interpolated value is made
     if (m_interpolationLocation == Mesh::Location::Edges)
     {
-        m_results.resize(m_mesh->GetNumEdges(), constants::missing::doubleValue);
+        m_edgeResults.resize(m_mesh->GetNumEdges(), constants::missing::doubleValue);
+        std::fill(m_edgeResults.begin(), m_edgeResults.end(), constants::missing::doubleValue);
```

Comment:
if `m_edgeResults` is uninitialized prior to resizing, there is no need to call `std::fill` on the next line.

Reply:
yes compute can be called multiple times